R/query_search.R

```diff
@@ -450,3 +450,26 @@ remote_report_metadata <- function(name, remote, config) {
   candidates$type <- "remote"
   candidates[c("id", "type", "path")]
 }
+
+
+build_query <- function(parameters) {
+  withCallingHandlers({
+    lapply(names(parameters), function(name) {
+      assert_scalar(parameters[[name]], name)
+    })
+  }, error = function(e) {
+    e$message <- paste0("Failed to build default query:\n", e$message)
+    stop(e)
+  })
+  quote_character <- function(x) {
+    out <- x
+    if (is.character(x)) {
+      out <- paste0('"', x, '"')
+    }
+    out
+  }
+  params <- lapply(parameters, quote_character)
```

Comment:
I had suggested

```
  params <- sprintf(dquote(list_to_character(lapply(parameters, as.character)))
```

but realise that this is even easier:

```suggestion
  params <- sprintf("parameter:%s == %s", names(parameters), names(parameters))
```

and then just paste that together with ` && `.

The query interface should do the validation of the parameters for us, we should not do the conversion to character etc here

Reply:
ooh ofc yes